allennlp/common/tee_logger.py

```diff
@@ -1,63 +0,0 @@
-"""
-A logger that maintains logs of both stdout and stderr when models are run.
-"""
-
-from typing import TextIO
-import os
-
-
-def replace_cr_with_newline(message: str):
-    """
-    TQDM and requests use carriage returns to get the training line to update for each batch
-    without adding more lines to the terminal output.  Displaying those in a file won't work
-    correctly, so we'll just make sure that each batch shows up on its one line.
-    :param message: the message to permute
-    :return: the message with carriage returns replaced with newlines
-    """
-    if "\r" in message:
-        message = message.replace("\r", "")
-        if not message or message[-1] != "\n":
-            message += "\n"
-    return message
-
-
-class TeeLogger:
-    """
-    This class is an attempt to maintain logs of both stdout and stderr for when models are run.
-    To use this class, at the beginning of your script insert these lines::
-
-        sys.stdout = TeeLogger("stdout.log", sys.stdout)
-        sys.stderr = TeeLogger("stdout.log", sys.stderr)
-    """
-
-    def __init__(
-        self, filename: str, terminal: TextIO, file_friendly_terminal_output: bool
-    ) -> None:
-        self.terminal = terminal
-        self.file_friendly_terminal_output = file_friendly_terminal_output
-        parent_directory = os.path.dirname(filename)
-        os.makedirs(parent_directory, exist_ok=True)
```

Comment:
This call is missing in your replacement function, which is why all the tests are failing. Just add this and then I think that should kill most of the CI output :)

Reply:
The actual issue was with calling `logging.shutdown`. According to the [docs](https://docs.python.org/3/library/logging.html?highlight=logging%20shutdown#logging.shutdown) the log handlers are properly flushed/closed by default at module exit. Hence it is not needed and the tests pass now. 

`TeeLogger's` API received a filepath and hence it was necessary to create necessary directories(`TeeLogger` doesn't assume `parent_directory` to be the `serialization_dir`) . But in the current change, everything is taken care in `prepare_global_logging` which assumes the directory is always `serialization_dir` and hence this need not be done.